torch_xla/csrc/data_ops.cpp

```diff
@@ -92,6 +92,18 @@ xla::XlaOp BuildView(xla::XlaOp input, absl::Span<const int64_t> output_sizes) {
   return XlaHelpers::DynamicReshape(input, complete_output_sizes);
 }
 
+xla::XlaOp SetDimensionSizes(xla::XlaOp input,
+                             absl::Span<const xla::XlaOp> output_sizes) {
+  XLA_CHECK_EQ(output_sizes.size(), XlaHelpers::ShapeOfXlaOp(input).rank());
+  for (int i = 0; i < output_sizes.size(); i++) {
+    xla::Shape dim_shape = XlaHelpers::ShapeOfXlaOp(output_sizes[i]);
+    if (dim_shape.is_dynamic()) {
```

Comment:
will this ever be true through? for example 
```
size = getDimenSize(tensor_a, 0);
```

even if dimension 0 of tensor_a is dynamic, size would still just be a scalar has a static shape. The value of it represent a dynamic shape but its own shape is static.

Reply:
Oh, I looked at your `SetSymIntNodeElements ` call, you can just check whether output_sizes is a `xla::Constant`. I think if it is a dynamic dimension it will be something else(need to verify..).

Reply:
I think you are right here. `output_sizes` could be either constants or `SizeNode`s.